### PR TITLE
Fix lint

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Fix lint errors introduced by #47 

#### Ticket Link
None

### CODE REVIEW

File: post.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ func (p *PostService) CreatePost(post *model.Post) error {
 		return normalizeAppErr(appErr)
 	}
 
+	//nolint:govet //lock value is not relevant to consumer
```
Comment (larkox) [inline post.go:22]: @lieut-data I wanted to have your attention on these two nolints. They are not introduced by my code, and I wanted to double check that the reasoning is in place.
The problem that the linter drops is:
> copylocks: assignment copies lock value to *post: github.com/mattermost/mattermost-server/v5/model.Post contains sync.RWMutex

File: experimental/freetextfetcher/store.go [hunk 2]
```diff
@@ -48,9 +48,9 @@ func (fts *freetextStore) StopFetching(userID string) error {
 	return fts.client.KV.Delete(fts.getKey(userID))
 }
 
-func (fts *freetextStore) ShouldProcessFreetext(userID, fetcherID string) (bool, string, error) {
+func (fts *freetextStore) ShouldProcessFreetext(userID, fetcherID string) (shouldProcess bool, payload string, err error) {
```
Comment (lieut-data) [inline experimental/freetextfetcher/store.go:51]: Curious what linting rule triggered this change? Named return values are usually exceptional.

File: post.go (hunk 1 above)
Comment (lieut-data) [reply post.go:22]: I think this was from an upstream change in mattermost-server -- I think we can handle this by using the new `Clone()` method on `model.Post`.

File: experimental/freetextfetcher/store.go (hunk 2 above)
Comment (hanzei) [reply experimental/freetextfetcher/store.go:51]: https://go-critic.github.io/overview#unnamedresult is the rule in question. Maybe enabling `opinionated` wasn't a good idea: https://github.com/mattermost/mattermost-plugin-api/blob/d4df2d1e8f88e7d892eac4f7aac83c87dd22dde8/.golangci.yml#L47

**Pull request merged**